packages/pentaho/src/Canvas/ToolbarTabs/ToolbarTabs.tsx

```diff
@@ -255,107 +240,82 @@ export const HvCanvasToolbarTabs = forwardRef<
-              value={selectedTab}
-              onChange={handleChangeSelectedTab}
-            >
-              {visibleTabs.map((tab, index) => (
+          <HvCanvasPanelTabs
+            classes={{ list: classes.tabsList }}
+            value={selectedTab}
+            onChange={handleChangeSelectedTab}
+          >
+            {visibleTabs.map((tab, index) => {
+              const btnSelected = selectedTab === tab.id;
+              return (
                 <HvCanvasPanelTab
+                  style={{
+                    width: tabWidth,
+                  }}
                   key={tab.id}
-                  ref={index === 0 ? tabRef : undefined}
                   id={String(tab.id)}
                   className={classes.tab}
                   value={tab.id}
-                  tabIndex={0}
-                  aria-label={tab.label}
+                  onKeyDown={(event) => handleKeyDownTab(event, tab.id)}
                 >
                   <div className={classes.tabContent}>
-                    <div
-                      className={cx({
-                        [classes.tabIcon]: !!tab.icon,
-                      })}
-                    >
-                      {tab.icon}
-                    </div>
-
-                    {/* Invisible components added in order for the tab width to go between max/min as some inner components are absolutely positioned */}
-                    <div role="none">{tab.label}</div>
-                    <div role="none">
-                      <Close />
+                    {tab.icon && (
+                      <div className={classes.tabIconContainer}>{tab.icon}</div>
+                    )}
+                    {!btnSelected || !allowTabEdit ? (
+                      <HvOverflowTooltip
+                        classes={{
+                          tooltipAnchor: classes.tabLabel,
+                        }}
+                        data={tab.label}
+                      />
+                    ) : (
+                      <ToolbarTabEditor
+                        classes={{
+                          label: cx(classes.tabLabel, classes.tabLabelEditor),
+                        }}
+                        value={tab.label}
+                        edit={isEditing}
+                        onEditChange={setIsEditing}
+                        onChange={(event, value) =>
+                          handleEdit(event, value, tab.id)
+                        }
+                        onBlur={(event, value) =>
+                          handleEdit(event, value, tab.id)
+                        }
+                        // We don't want the arrow keys to trigger the tab navigation
+                        onKeyDown={(e) => e.stopPropagation()}
+                      />
+                    )}
+                    <div className={classes.closeIconContainer}>
+                      <CloseXS
+                        aria-hidden
+                        data-testid="delete-icon" // For testing purposes
```

Comment:
Icons have a `data-name` that we can use for assertions ([TagsInput example](https://github.com/lumada-design/hv-uikit-react/blob/master/packages/core/src/TagsInput/TagsInput.test.tsx#L41)). We would need to `getByRole("tab")` first though

Reply:
I'll address this in another PR